runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -75,21 +69,52 @@ private static void bootstrapLayoutServer(IClientRouter router, Layout layout)
      * @param router Router connected to the node.
      * @param layout Layout to bootstrap with
      */
-    private static void bootstrapManagementServer(IClientRouter router, Layout layout)
-            throws ExecutionException, InterruptedException, AlreadyBootstrappedException {
+    public static void bootstrapManagementServer(IClientRouter router, Layout layout) {
         ManagementClient managementClient
                 = new ManagementClient(router, layout.getEpoch(), layout.getClusterId());
 
-        try {
-            CFUtils.getUninterruptibly(managementClient.bootstrapManagement(layout),
-                    AlreadyBootstrappedException.class);
-        } catch (AlreadyBootstrappedException abe) {
-            if (!managementClient.getLayout().get().equals(layout)) {
-                log.error("BootstrapUtil: Management Server {}:{} already bootstrapped with "
-                        + "different layout.", router.getHost(), router.getPort());
-                throw abe;
-            }
-        }
+        bootstrap(router, layout,
+                () -> managementClient.bootstrapManagement(layout),
+                managementClient::getLayout);
+    }
+
+    /**
+     *
+     * Bootstrap a server with a specific function.
+     *
+     * @param router          Router connected to the node.
+     * @param layout          Layout to bootstrap with.
+     * @param bootstrapFuture A supplier for the function to bootstrap a server.
+     * @param getLayoutFuture A supplier for the function to get a layout.
+     */
+    private static void bootstrap(IClientRouter router, Layout layout,
+                                 Supplier<CompletableFuture<Boolean>> bootstrapFuture,
+                                 Supplier<CompletableFuture<Layout>> getLayoutFuture) {
```

Comment:
let's rename `getLayoutFuture` to something like: `layoutFutureProvider`
same for bootstrapFuture

Reply:
fixed